tests/Functional/ORMModelFactoryTest.php

```diff
@@ -622,6 +622,26 @@ public function can_find_or_create_entity_with_enum(): void
         self::assertSame($entityWithEnum->object(), $entityWithEnum2->object());
     }
 
+    /**
+     * @test
+     */
+    public function can_use_virtual_properties_as_attributes(): void
+    {
+        $category = CategoryFactory::createOne([
+            'posts' => PostFactory::new(['virtualTitle' => 'regular post'])->many(2),
+            'additionalPosts' => PostFactory::new(['virtualTitle' => 'additional post'])->many(2),
+            'addPost' => PostFactory::new(['virtualTitle' => 'regular post added']),
+        ]);
+
+        self::assertCount(5, $category->getPosts());
+
+        self::assertSame('additional post', $category->getPosts()[0]->getTitle());
```

Comment:
there is a small glitch here: `posts` will be hydrated with post persist callback, then they will be added at the last position in the collection